framework/src/main/java/io/github/kgress/scaffold/webdriver/WebDriverManager.java

```diff
@@ -33,530 +53,765 @@
+    var deviceType = Optional.ofNullable(getDesiredCapabilities().getMobile().getDeviceType());
+    var deviceOrientation = Optional
+        .ofNullable(getDesiredCapabilities().getMobile().getDeviceOrientation());
+    caps.setCapability("appium:platformVersion", platformVersion);
+    deviceOrientation.ifPresent(
+        orientation -> caps.setCapability("deviceOrientation", orientation.getDeviceOrientation()));
+    deviceType.ifPresent(type -> caps.setCapability("deviceType", type.getDeviceType()));
+
+    return caps;
+  }
+
+  /**
+   * This helper method configures a {@link RemoteWebDriver}. A {@link RemoteWebDriver} is used for
+   * testing against Selenium Grid or SauceLabs.
+   * <p>
+   * Serves the purpose of configuring a remote browser for both desktop and mobile emulation. Since
+   * mobile emulation uses {@link RunType#SAUCE_MOBILE_EMULATOR}, we don't need to worry about any
+   * desktop specific settings being applied to the execution.
+   *
+   * @param browserOptions the browser configuration to be used with the new {@link
+   *                       RemoteWebDriver}
+   * @return the new {@link RemoteWebDriver} as a {@link WebDriver}
+   */
+  private WebDriver configureRemoteBrowser(MutableCapabilities browserOptions,
+      String testName) {
+    RemoteWebDriver remoteWebDriver;
+    var runType = getDesiredCapabilities().getRunType(); // already null checked through lombok
+    // These values are already null checked previously. However, using optionals for their lambdas
+    // cut down on code and it looks prettier. So let's do that!
+    var desktopRunPlatform = Optional.ofNullable(getDesiredCapabilities().getRunPlatform());
+    var desktopBrowserType = Optional.ofNullable(getDesiredCapabilities().getBrowserType());
+    var desktopBrowserVersion = Optional.ofNullable(getDesiredCapabilities().getBrowserVersion());
+
+    // Sauce capability keys are different and must be set accordingly/.
+    // Only set if not sauce mobile emulator
+    if (runType != SAUCE_MOBILE_EMULATOR) {
+      if (runType == SAUCE) {
+        desktopBrowserVersion
+            .ifPresent(version -> browserOptions.setCapability("browserVersion", version));
+        desktopBrowserType.ifPresent(
+            browser -> browserOptions.setCapability("browserName", browser.getBrowserName()));
+        desktopRunPlatform.ifPresent(
+            platform -> browserOptions.setCapability("platformName", platform.getPlatform()));
+      } else {
+        desktopBrowserVersion
+            .ifPresent(version -> browserOptions.setCapability("version", version));
+        desktopRunPlatform.ifPresent(
+            platform -> browserOptions.setCapability("platform", platform.getPlatform()));
+      }
+    }
+    // Create the remote web driver and set the session id, adding it to the test context
+    remoteWebDriver = createRemoteWebDriver(browserOptions, testName);
+    //    private Set<Cookie> cookieJar = new TreeSet<>();
```

Comment:
Leftover comment?

Reply:
oo yea that's leftover. I'll take care of that